src/pages/Login.jsx

```diff
@@ -1,9 +1,92 @@
+/* eslint-disable no-return-assign */
+/* eslint-disable consistent-return */
+/* eslint-disable react-hooks/exhaustive-deps */
+import { useState, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { ToastContainer } from 'react-toastify';
+import { Link, useNavigate } from 'react-router-dom';
+import Button from '../components/Button/Button';
+import { userloginFields } from '../constants/formFields';
+import Input from '../components/Auth/Input';
+import Loginphoto from '../assets/login-illustration.svg';
+import GoogleButton from '../components/Button/GoogleButton';
+import { login } from '../redux/actions/auth.loginActions';
+import 'react-toastify/dist/ReactToastify.css';
+
+const fieldState = {};
+userloginFields.forEach((field) => (fieldState[field.id] = ''));
+
 const Login = () => {
+	const navigate = useNavigate();
+	const dispatch = useDispatch();
+	const { user, loading } = useSelector((state) => state.login);
+	const [loginState, setLoginState] = useState(fieldState);
+
+	const handleChange = (e) =>
+		setLoginState({ ...loginState, [e.target.id]: e.target.value });
+
+	const handleLogin = (e) => {
+		e.preventDefault();
+		dispatch(login(loginState));
+	};
+
+	useEffect(() => {
+		setTimeout(() => {
+			if (user?.data?.others?.RoleId === 1) return navigate('/admin');
```

Comment:
- better to use `redirect` than this `useNavigate` hook.

https://reactrouter.com/en/main/fetch/redirect